app/Console/Commands/DeepCleanDatabase.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace App\Console\Commands;
+
+use App\Utils\DatabaseCleanupUtils;
+use Illuminate\Console\Command;
+use Illuminate\Support\Facades\DB;
+
+class DeepCleanDatabase extends Command
+{
+    /**
+     * The name and signature of the console command.
+     */
+    protected $signature = 'db:deep-clean';
+
+    /**
+     * The console command description.
+     */
+    protected $description = 'Prune unused records from the CDash database';
+
+    /**
+     * Execute the console command.
+     */
+    public function handle(): void
+    {
+        $num_deleted = DB::delete("DELETE FROM banner WHERE projectid != 0 AND
+                projectid NOT IN (SELECT id FROM project)");
```

Comment:
As discussed in https://github.com/Kitware/CDash/pull/2552, `WHERE NOT EXISTS ...` is probably faster for Postgres, and shouldn't be any slower on MySQL.  Apply throughout.